#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_debian-patch.dpatch by Junichi Uekawa <dancer@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Debian modifications to ladspa-sdk

@DPATCH@

--- ladspa-sdk-1.1.orig/src/makefile
+++ ladspa-sdk-1.1/src/makefile
@@ -5,9 +5,9 @@
 # Change these if you want to install somewhere else. In particularly
 # you may wish to remove the middle "local/" part of each entry.
 
-INSTALL_PLUGINS_DIR	=	/usr/local/lib/ladspa/
-INSTALL_INCLUDE_DIR	=	/usr/include/
-INSTALL_BINARY_DIR	=	/usr/local/bin/
+INSTALL_PLUGINS_DIR	?=	/usr/local/lib/ladspa/
+INSTALL_INCLUDE_DIR	?=	/usr/include/
+INSTALL_BINARY_DIR	?=	/usr/local/bin/
 
 ###############################################################################
 #
@@ -16,7 +16,7 @@
 
 INCLUDES	=	-I.
 LIBRARIES	=	-ldl -lm
-CFLAGS		=	$(INCLUDES) -Wall -Werror -O3 -fPIC
+CFLAGS		=	$(INCLUDES) -O3 -fPIC
 CXXFLAGS	=	$(CFLAGS)
 PLUGINS		=	../plugins/amp.so				\
 			../plugins/delay.so				\
@@ -36,11 +36,11 @@
 
 ../plugins/%.so:	plugins/%.c ladspa.h
 	$(CC) $(CFLAGS) -o plugins/$*.o -c plugins/$*.c
-	$(LD) -o ../plugins/$*.so plugins/$*.o -shared
+	$(LD) -lc $(LIBRARIES) -o ../plugins/$*.so plugins/$*.o -shared
 
 ../plugins/%.so:	plugins/%.cpp ladspa.h
 	$(CPP) $(CXXFLAGS) -o plugins/$*.o -c plugins/$*.cpp
-	$(CPP) -o ../plugins/$*.so plugins/$*.o -shared
+	$(CPP) $(LIBRARIES) -o ../plugins/$*.so plugins/$*.o -shared
 
 ###############################################################################
 #
@@ -59,7 +59,7 @@
 	play /tmp/test.wav
 
 install:	targets
-	-mkdirhier	$(INSTALL_PLUGINS_DIR)				\
+	-mkdir -p	$(INSTALL_PLUGINS_DIR)				\
 			$(INSTALL_INCLUDE_DIR)				\
 			$(INSTALL_BINARY_DIR)
 	cp ../plugins/* $(INSTALL_PLUGINS_DIR)
--- ladspa-sdk-1.1.orig/src/search.c
+++ ladspa-sdk-1.1/src/search.c
@@ -99,10 +99,10 @@
 
   pcLADSPAPath = getenv("LADSPA_PATH");
   if (!pcLADSPAPath) {
+    pcLADSPAPath="/usr/lib/ladspa/";
     fprintf(stderr,
 	    "Warning: You do not have a LADSPA_PATH "
 	    "environment variable set.\n");
-    return;
   }
   
   pcStart = pcLADSPAPath;
--- ladspa-sdk-1.1.orig/src/load.c
+++ ladspa-sdk-1.1/src/load.c
@@ -54,6 +54,8 @@
        to search. */
 
     pcLADSPAPath = getenv("LADSPA_PATH");
+    if (!pcLADSPAPath)
+      pcLADSPAPath="/usr/lib/ladspa";
     
     if (pcLADSPAPath) {
 
